Purchases/Public/RCPurchases.m

```diff
@@ -197,64 +200,62 @@ - (RCPurchaserInfo *)readPurchaserInfoFromCache {
 }
 
 - (void)cachePurchaserInfo:(RCPurchaserInfo *)info {
-    if (info.JSONObject) {
-        NSError *jsonError = nil;
-        NSData *jsonData = [NSJSONSerialization dataWithJSONObject:info.JSONObject
-                                                           options:0
-                                                             error:&jsonError];
-        if (jsonError == nil) {
-            [self.userDefaults setObject:jsonData
-                                  forKey:self.purchaserInfoUserDefaultCacheKey];
-        }
+    if (info) {
+        [self dispatch:^{
+            if (info.JSONObject) {
+                NSError *jsonError = nil;
+                NSData *jsonData = [NSJSONSerialization dataWithJSONObject:info.JSONObject
+                                                                   options:0
+                                                                     error:&jsonError];
+                if (jsonError == nil) {
+                    [self.userDefaults setObject:jsonData
+                                          forKey:self.purchaserInfoUserDefaultCacheKey];
+                }
+            }
+            self.cachesLastUpdated = [NSDate date];
+        }];
     }
 }
 
+- (BOOL)isCacheStale {
```

Comment:
This should be `isCacheValid`. Or change the return to !(self.cachesLastUpdated != nil && timeSinceLastCheck < 60. * 5). But the return of this method is true when it is not stale

Reply:
😺